### PR TITLE
Add snippets for cmake

### PR DESCRIPTION
**Pull request by burstknight**
I add snippets for cmake.

### CODE REVIEW

File: snippets/cmake.json [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+		"prefix": "cmr",
+		"body": [
+			"cmake_minimum_required(VERSION ${1:3.16})$2"
+		],
+		"description": "Add the snippet for cmake_minimum_required()"
+	},
+	"message()": {
+		"prefix": "msg",
+		"body": [
+			"message(${1|STATUS,DEBUG,WARNING,SEND_ERROR_FATAL_ERROR|} \"${2:message}\")$3"
```
Comment (nega0) [inline snippets/cmake.json:46]: shouldn't this be `SEND_ERROR, FATAL_ERROR` ?

File: snippets/cmake.json [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+			"if(NOT CMAKE_BUILD_TYPE)",
+			"\tset(default_build_type \"Debug\")",
+			"\tmessage(STATUS \"Set the build type to `\\${default_build_type}` as none was specified.\")",
+			"\tset(CMAKE_BUILD_TYPE \\${default_build_type} CACHE STRING \"Chooce the build type.\" FORCE)",
+			"set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS",
+			"\t\"Debug\" \"Release\" \"MinSizeRel\" \"RelWithDebInfo\")",
+			"endif()",
+			"message(STATUS \"$2 Build Type: \\${CMAKE_BUILD_TYPE}\")",
+			"",
+			"# Set the verson for $2",
```
Comment (nega0) [inline snippets/cmake.json:17]: "version"

File: snippets/cmake.json [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+		"prefix": "cmi",
+		"body": [
+			"cmake_minimum_required(VERSION ${1:3.16})",
+			"project(${2:myProject})",
+			"",
+			"if(NOT CMAKE_BUILD_TYPE)",
+			"\tset(default_build_type \"Debug\")",
+			"\tmessage(STATUS \"Set the build type to `\\${default_build_type}` as none was specified.\")",
+			"\tset(CMAKE_BUILD_TYPE \\${default_build_type} CACHE STRING \"Chooce the build type.\" FORCE)",
+			"set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS",
```
Comment (nega0) [inline snippets/cmake.json:12]: missing an indent here

File: snippets/cmake.json [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+			"message(STATUS \"\\${PROJECT_NAME} version: \\${PROJECT_VERSION}\")"
+		],
+		"description": "Init 'CMakeLists.txt'."
+	},
+	"project()": {
+		"prefix": "prj",
+		"body":[
+			"project(${1:myProject})$2"
+		],
+		"description": "Add the snippet for prject()"
```
Comment (nega0) [inline snippets/cmake.json:34]: "project" 😉

File: snippets/cmake.json [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+			"message(${1|STATUS,DEBUG,WARNING,SEND_ERROR_FATAL_ERROR|} \"${2:message}\")$3"
+		],
+		"description": "Add the snippet for message()"
+	},
+	"export compile commands": {
+		"prefix": "ecc",
+		"body": [
+			"set(CMAKE_EXPORT_COMPILE_COMMANDS ON)"
+		],
+		"description": "Add teh snippet to generate 'compile_commands.json'"
```
Comment (nega0) [inline snippets/cmake.json:55]: "the" 😉

**Comment by burstknight** [comment]
Thank you, @nega0. I fixed them.

**Review by OkelleyDevelopment — APPROVED** [review]

**Pull request merged**